### ISSUE TITLE
신세벌식 P2 쌍기역 받침 입력이 안 됩니다.

### ISSUE THREAD

**Issue by demokritos** [Riey/kime#646]
받침 ㄱ을 두 번 입력하면 받침 ㄲ이 입력돼야 하는데 처음 받침 ㄱ으로만 음절 하나 완성하고 다음에 받침 ㄱ을 따로 입력하네요.
예로 '겪을'을 입력하다면 '격ㄱ을'이 입력됩니다.
아치 리눅스 그놈  45.0 쓰고 있고 kime는 aur/kime-git를 설치하였고, 버전 3.0.2.r6.g9dd6411-1입니다.

**Comment by Riey** [comment]
https://github.com/Riey/kime/blob/develop/docs/CONFIGURATION.ko.md#composejongseongssang

이 애드온을 켜시면 아마 받침이 정상적으로 합쳐지실겁니다

**Comment by Riey** [comment]
그런데 세벌식에선 쌍자음 받침을 이렇게 입력하는게 보통인가요? 그렇다면 해당 애드온을 디폴트로 켜야할 필요는 있을것 같네요.

**Comment by demokritos** [comment]
애드온이란 게 추가적으로 기능을 넣는 건데 제가 추가적으로 받침 입력하는 방법을 요구하는 게 아니라 신세벌식 P2에서는 쌍기역 받침을 넣을 수 없습니다. 신세벌식은 쉬프트키를 안 쓰려고 하는 세벌식인 것 같은데 그래서 키 수가 적어서 모든 받침을 단일 키로 넣을 수 없습니다. 그냥 쌍시옷 받침을 많이 쓰니까 그것만 단일 키로 할당한 것 같습니다. 모든 받침이 단일 키로 할당된 다른 세벌식도 libhangul에서는 받침 키 조합해서 넣을 수 있네요. 세벌식은 초중종 세벌로 입력하는 거니까 종성과 초성을 혼동할 수가 없기 때문에 이건 당연한 것 같습니다.

**Comment by Riey** [comment]
> 애드온이란 게 추가적으로 기능을 넣는 건데 제가 추가적으로 받침 입력하는 방법을 요구하는 게 아니라 신세벌식 P2에서는 쌍기역 받침을 넣을 수 없습니다. 신세벌식은 쉬프트키를 안 쓰려고 하는 세벌식인 것 같은데 그래서 키 수가 적어서 모든 받침을 단일 키로 넣을 수 없습니다. 그냥 쌍시옷 받침을 많이 쓰니까 그것만 단일 키로 할당한 것 같습니다. 모든 받침이 단일 키로 할당된 다른 세벌식도 libhangul에서는 받침 키 조합해서 넣을 수 있네요. 세벌식은 초중종 세벌로 입력하는 거니까 종성과 초성을 혼동할 수가 없기 때문에 이건 당연한 것 같습니다.

네 설명을 조금 드리자면 kime은 여러 자판을 지원하지만 코드는 동일하다보니 원래 당연히 있어야 하는 기능도 분류상으론 `애드온`으로 돼있는 경우가 있습니다.
이런 경우 해당 자판에는 이 기능을 기본 `애드온`으로 설정을 해놓습니다 두벌식의 경우에도 입력에 필수인 종성이 초성으로 바뀌는 기능이 애드온으로 되어있죠.
지금은 그런 필수적인 기능이 빠져있었던것 같습니다.

**Comment by opqriu** [comment]
> 그런데 세벌식에선 쌍자음 받침을 이렇게 입력하는게 보통인가요? 그렇다면 해당 애드온을 디폴트로 켜야할 필요는 있을것 같네요.


네. 

세벌식은 원래 쌍자음 쌍모음을 하나의 키를 두 번 연속 입력하는 방식을 많이 씁니다. 제가 390 써도, 저는 항상 한 키를 두 번 연속 입력해요. 왜냐면 쉬프트는 약지의 피곤함이 크거든요. 390이든 P2든 마찬가지에요. P2가 더 연속타가 많긴 하지만요.

그나저나 kime에 신세벌p2가 있었군요. 공식 지원하는게 없어서 390 쓴지 오랜데... 오오오 감사 ㅜ.ㅜ

**Comment by opqriu** [comment]
config에 ComposeChoseongSsang가 all로 기본으로 켜져 있는 것 같은데요. 신세벌P2에서만 안 먹히는 거 같아요.

**Comment by opqriu** [comment]
감사합니다. 390과 신세벌식P2 모두 잘 작동하네요. 건의 드리자면 세벌식은 390 391 P2 모두 ComposeJungseongSsang 옵션을 기본으로 주시면 좋을거 같아요 ^^ 세벌식은 원래 이렇게 작동해요~

**Comment by opqriu** [comment]
감사합니다. 잘 작동하네요.

**Comment by Riey** [comment]
늦었지만 [#752](https://github.com/Riey/kime/pull/752)에서 신세벌식 P2의 기본 애드온에 `ComposeJongseongSsang`을 추가했습니다. 이제 별도 설정 없이 받침 ㄱ을 두 번 눌러 ㄲ 받침을 입력할 수 있고, 말씀해주신 받침 ㅅ 두 번 → ㅆ도 같은 애드온으로 동작합니다. 회귀 방지 테스트(`sebeolsik_3sin_p2.rs`)도 함께 추가했습니다.

이 수정은 3.2.0에 포함되어 릴리스되었습니다. 제보해주셔서 감사합니다!
